source/fuzz/fuzzer_pass_flatten_conditional_branches.h

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef SOURCE_FUZZ_FUZZER_PASS_FLATTEN_CONDITIONAL_BRANCHES_H
+#define SOURCE_FUZZ_FUZZER_PASS_FLATTEN_CONDITIONAL_BRANCHES_H
+
+#include "source/fuzz/fuzzer_pass.h"
+
+namespace spvtools {
+namespace fuzz {
+
+class FuzzerPassFlattenConditionalBranches : public FuzzerPass {
+ public:
+  FuzzerPassFlattenConditionalBranches(
+      opt::IRContext* ir_context, TransformationContext* transformation_context,
+      FuzzerContext* fuzzer_context,
+      protobufs::TransformationSequence* transformations);
+
+  ~FuzzerPassFlattenConditionalBranches() override;
+
+  void Apply() override;
+
+  // Returns the nesting depth of the given block, i.e. the number of constructs
+  // containing it. Headers and merge blocks of constructs are not considered
+  // part of a construct.
+  static uint32_t NestingDepth(opt::IRContext* ir_context, uint32_t block_id);
```

Comment:
Could this be in the private section?

Reply:
OK, I see it is public because you have tests for it.

I think it would be best for this to live in structured cfg analysis, as it's generally useful.

You could either do a PR there, first, or you could open an issue and put a TODO comment here, get this PR merged, and then do a refactoring.

The tests should migrate to the tests for structured CFG analysis.

Reply:
I have opened PR #3771 .